src/collectors.cc

```diff
@@ -519,7 +521,7 @@ Int SingleCollectWord(Obj sc, Obj vv, Obj w)
             /* collect a whole word exponent pair                          */
             else if( *pw == *nw && INT_INTOBJ(avc[gn]) == gn ) {
               gn = SAddWordIntoExpVec(
-                   v, *pw, *ge, ebits, expm, ro, pow, lpow  );
+                   v, *nw, *lw, *ge, ebits, expm, ro, pow, lpow  );
```

Comment:
Note that by the check in the `else if` above, `*pw == *nw`; and in `SC_PUSH_WORD` we set `*lw` to point at the end of the word, as required for this call.